designer/server/src/main/scala/pl/touk/nussknacker/ui/api/StickyNotesApiHttpService.scala

```diff
@@ -0,0 +1,230 @@
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class StickyNotesApiHttpService(
+    authManager: AuthManager,
+    stickyNotesRepository: StickyNotesRepository,
+    scenarioService: ProcessService,
+    scenarioAuthorizer: AuthorizeProcess,
+    dbioActionRunner: DBIOActionRunner,
+    stickyNotesSettings: StickyNotesSettings
+)(implicit executionContext: ExecutionContext)
+    extends BaseHttpService(authManager)
+    with LazyLogging {
+
+  private val securityInput = authManager.authenticationEndpointInput()
+
+  private val endpoints = new StickyNotesApiEndpoints(securityInput)
+
+  expose {
+    endpoints.stickyNotesGetEndpoint
+      .serverSecurityLogic(authorizeKnownUser[StickyNotesError])
+      .serverLogicEitherT { implicit loggedUser =>
+        { case (scenarioName, versionId) =>
+          for {
+            scenarioId      <- getScenarioIdByName(scenarioName)
+            _               <- isAuthorized(scenarioId, Permission.Read)
+            processActivity <- fetchStickyNotes(scenarioId, versionId)
+          } yield processActivity.toList
+        }
+      }
+  }
+
+  expose {
+    endpoints.stickyNotesAddEndpoint
+      .serverSecurityLogic(authorizeKnownUser[StickyNotesError])
+      .serverLogicEitherT { implicit loggedUser =>
+        { case (scenarioName, requestBody) =>
+          for {
+            scenarioId      <- getScenarioIdByName(scenarioName)
+            _               <- isAuthorized(scenarioId, Permission.Write)
+            count           <- getStickyNotesCount(scenarioId, requestBody.scenarioVersionId)
+            _               <- validateStickyNotesCount(count, stickyNotesSettings)
+            _               <- validateStickyNoteContent(requestBody.content, stickyNotesSettings)
+            processActivity <- addStickyNote(scenarioId, requestBody)
+          } yield processActivity
+        }
+      }
+  }
+
+  expose {
+    endpoints.stickyNotesUpdateEndpoint
+      .serverSecurityLogic(authorizeKnownUser[StickyNotesError])
+      .serverLogicEitherT { implicit loggedUser =>
+        { case (scenarioName, requestBody) =>
+          for {
+            scenarioId      <- getScenarioIdByName(scenarioName)
+            _               <- isAuthorized(scenarioId, Permission.Write)
+            _               <- validateStickyNoteContent(requestBody.content, stickyNotesSettings)
+            processActivity <- updateStickyNote(requestBody)
+          } yield processActivity.toInt
+        }
+      }
+  }
+
+  expose {
+    endpoints.stickyNotesDeleteEndpoint
+      .serverSecurityLogic(authorizeKnownUser[StickyNotesError])
+      .serverLogicEitherT { implicit loggedUser =>
+        { case (scenarioName, noteId) =>
+          for {
+            scenarioId      <- getScenarioIdByName(scenarioName)
+            _               <- isAuthorized(scenarioId, Permission.Write)
+            processActivity <- deleteStickyNote(noteId)
+          } yield processActivity.toInt
```

Comment:
processActivity is already an Int

Reply:
That is funny but I've added it to prevent 
```
[error]...
StickyNotesApiHttpService.scala:102:19: a pure expression does nothing in statement position
[error]           } yield processActivity
```  
When you remove `.toInt` code does not compile (idea see no problem though). Scala gives us warning and we must be handling it as error :shrug: